There are a few warnings when building without llamafile:
```
ggml.c: In function ‘ggml_compute_forward_mul_mat’:
ggml.c:12040:18: warning: unused variable ‘row_size’ [-Wunused-variable]
12040 |     const size_t row_size = ggml_row_size(vec_dot_type, ne10);
      |                  ^~~~~~~~
ggml.c:12039:18: warning: unused variable ‘wdata’ [-Wunused-variable]
12039 |     const void * wdata    = (src1->type == vec_dot_type) ? src1->data : params->wdata;
      |                  ^~~~~
ggml.c:11906:19: warning: unused variable ‘r3’ [-Wunused-variable]
11906 |     const int64_t r3 = ne13/ne03;
      |                   ^~
ggml.c:11905:19: warning: unused variable ‘r2’ [-Wunused-variable]
11905 |     const int64_t r2 = ne12/ne02;
      |                   ^~
ggml.c:11883:16: warning: unused variable ‘src1_cont’ [-Wunused-variable]
11883 |     const bool src1_cont = ggml_is_contiguous(src1);
      |
```

--- a/ggml.c
+++ b/ggml.c
@@ -11880,8 +11880,6 @@ static void ggml_compute_forward_mul_mat(
 
     const enum ggml_type type = src0->type;
 
-    const bool src1_cont = ggml_is_contiguous(src1);
-
     enum ggml_type    const vec_dot_type          = type_traits[type].vec_dot_type;
     ggml_from_float_t const from_float_to_vec_dot = type_traits[vec_dot_type].from_float;
     int64_t           const vec_dot_num_rows      = type_traits[type].nrows;
@@ -11900,10 +11898,6 @@ static void ggml_compute_forward_mul_mat(
     GGML_ASSERT(nb0 <= nb1);
     GGML_ASSERT(nb1 <= nb2);
     GGML_ASSERT(nb2 <= nb3);
-
-    // broadcast factors
-    const int64_t r2 = ne12/ne02;
-    const int64_t r3 = ne13/ne03;
 
     // nb01 >= nb00 - src0 is not transposed
     //   compute by src0 rows
@@ -11985,6 +11979,12 @@ static void ggml_compute_forward_mul_mat(
 #endif
 
 #if GGML_USE_LLAMAFILE
+    const bool src1_cont = ggml_is_contiguous(src1);
+
+    // broadcast factors
+    const int64_t r2 = ne12 / ne02;
+    const int64_t r3 = ne13 / ne03;
+
     if (src1_cont) {
         for (int64_t i13 = 0; i13 < ne13; i13++)
             for (int64_t i12 = 0; i12 < ne12; i12++)
@@ -12036,11 +12036,11 @@ UseGgmlGemm1:;
         return;
     }
 
-    const void * wdata    = (src1->type == vec_dot_type) ? src1->data : params->wdata;
-    const size_t row_size = ggml_row_size(vec_dot_type, ne10);
-
 #if GGML_USE_LLAMAFILE
     if (src1->type != vec_dot_type) {
+        const void* wdata = (src1->type == vec_dot_type) ? src1->data : params->wdata;
+        const size_t row_size = ggml_row_size(vec_dot_type, ne10);
+
         for (int64_t i13 = 0; i13 < ne13; i13++)
             for (int64_t i12 = 0; i12 < ne12; i12++)
                 if (!llamafile_sgemm(ne01, ne11, ne00/ggml_blck_size(src0->type),

--- a/ggml.c
+++ b/ggml.c
@@ -11899,6 +11899,12 @@ static void ggml_compute_forward_mul_mat(
     GGML_ASSERT(nb1 <= nb2);
     GGML_ASSERT(nb2 <= nb3);
 
+    // broadcast factors
+    const int64_t r2 = ne12 / ne02;
+    const int64_t r3 = ne13 / ne03;
+    UNUSED(r2);
+    UNUSED(r3);
+
     // nb01 >= nb00 - src0 is not transposed
     //   compute by src0 rows
 
@@ -11980,10 +11986,6 @@ static void ggml_compute_forward_mul_mat(
 
 #if GGML_USE_LLAMAFILE
     const bool src1_cont = ggml_is_contiguous(src1);
-
-    // broadcast factors
-    const int64_t r2 = ne12 / ne02;
-    const int64_t r3 = ne13 / ne03;
 
     if (src1_cont) {
         for (int64_t i13 = 0; i13 < ne13; i13++)

--- a/ggml.c
+++ b/ggml.c
@@ -19683,7 +19683,7 @@ static void ggml_graph_compute_thread_sync_task(int * task_phase, struct ggml_co
         * task_phase = atomic_load(&state->shared->node_task);
         if (* task_phase != last_task_phase) break;
 #if defined(__SSE3__)
-        //Tell the processor we're spinning.  It's a processor hint for spinlocks.
+        // Tell the processor we're spinning.  It's a processor hint for spinlocks.
         _mm_pause();
 #endif
     }

